pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -152,18 +170,91 @@ func parseURL(input string) (url.URL, error) {
 	return *uri, nil
 }
 
+// getV1Registries transforms the registries arrays of v1 into an array of
+// v2 registries of type Registry.
+func getV1Registries(config *tomlConfig) ([]Registry, error) {
+	regs := make(map[string]Registry)
+
+	makeRegistry := func(s string) (Registry, error) {
+		var reg Registry
+		url, err := parseURL(s)
+		if err != nil {
+			return reg, err
+		}
+		reg.URL = url
+		reg.Prefix = stripURIScheme(url)
+		return reg, nil
+	}
```

Comment:
This could be simpler, roughly like:

```go
regs := make(map[string]*Registry)
getRegistry := func(s string) (*Registry, error) { // Note: _pointer_ to a long-lived object
    prefix = …
    reg, exists := regs[prefix]
    if !exists {
        reg := &Registry{}
        // initialize reg
        regs[prefix] = reg
    }
   return reg, nil
}
…
for _, search := range config.V1Registries.Search.Registries {
    reg, err := getRegistry(search)
    if err != nil {
        return []Registry{}, err
    }
    reg.Search = true
}
```

notably all of the `||` updates go away, and we don’t need to worry about what parts of `reg` have been discarded in the `exists` case of `addRegistry`.

Reply:
Thanks for your great review. This one made the code _much_ simpler and more readable.